extensions/amp-animation/0.1/web-animations-polyfill.js

```diff
@@ -26,8 +26,21 @@ const POLYFILLED = '__AMP_WA';
  * @param {!Window} win
  */
 function forceOnSafari(win) {
-  if (Services.platformFor(win).isSafari()) {
-    win.document.documentElement.animate = () => /** @type {!Animation} */ ({});
+  const {prototype} = win.Animation || {};
+  if (prototype && Services.platformFor(win).isSafari()) {
+    const animationUnimplemented = {};
+    for (const k in prototype) {
+      try {
+        if (typeof prototype[k] === 'function') {
+          animationUnimplemented[k] = () => {};
+        }
+      } catch (unusedError) {
+        // Non-method member access (like .id) will fail.
+        // We don't care.
+      }
+    }
+    win.document.documentElement.animate = () =>
+      /** @type {!Animation} */ (animationUnimplemented);
```

Comment:
Missed it the last time, but I think it's critical: `forceOnSafari(win);` should be inside `if (!win[POLYFILLED]) {`, no?

Reply:
True!